lib/units/validateURI.js

```diff
@@ -1,14 +1,60 @@
+const deepEqual = require('deep-equal');
+
 const APIARY_URI_TYPE = 'text/vnd.apiary.uri';
 
+/**
+ * Parses a given query string into an Object.
+ * @param {string} queryString
+ */
+const parseQueryString = (queryString) => {
```

Comment:
Why DIY and not https://nodejs.org/api/url.html ?

Reply:
My biggest concern was that using `URL` class you cannot create relative urls:

```js
new URL('/admin')
// Exception: invalid URL
```

> There is a fair chance my unfamiliarity with the API gets in a way, so I'll learn more and conclude whether we can use native API for URL parsing.

It requires a `base`, which may mislead if present in the comparison diff.

There is a `url` module that can parse relative urls, but it's [marked as legacy](https://nodejs.org/api/url.html#url_legacy_url_api). The WHATWG URL API (mentioned above) is suggested to be used instead.

Reply:
Using

```
const url = require('url')
``

replaces the previous functionality.

Reply:
Changed in fcd0ccf.